src/main/java/org/folio/services/lines/PoLinesService.java

```diff
@@ -400,25 +462,27 @@ private void populateTitleBasedOnPackagePoLine(Title title, PoLine packagePoLine
     }
   }
 
-  private Future<Tx<PoLine>> upsertTitle(Tx<PoLine> poLineTx, DBClient client) {
+  private Future<PoLine> updateTitle(Conn conn, PoLine poLine) {
     Promise<Results<Title>> promise = Promise.promise();
-    PoLine poLine = poLineTx.getEntity();
     Criterion criterion = getCriteriaByFieldNameAndValueNotJsonb(POLINE_ID_FIELD, poLine.getId());
 
-    client.getPgClient().get(poLineTx.getConnection(), TITLES_TABLE, Title.class, criterion, true, false, promise);
+    conn.get(TITLES_TABLE, Title.class, criterion, true)
+      .onComplete(promise);
+
     return promise.future()
```

Comment:
it looks like promise variable is not needed anymore. You could just use 
`conn.get(TITLES_TABLE, Title.class, criterion, true).compose(...)
`
Line `.recover(Future::failedFuture);` does nothing and can be removed

Reply:
Fixed, thank you